falcon/media/multipart.py

```diff
@@ -0,0 +1,402 @@
+import cgi
+import re
+from urllib.parse import unquote_to_bytes
+
+from falcon import errors
+from falcon import request_helpers
+from falcon.media.base import BaseHandler
+from falcon.media.handlers import Handlers
+from falcon.util import BufferedStream
+from falcon.util import misc
+
+# TODO(vytas):
+#   * Better support for form-wide charset setting
+#   * Clean up, simplify, and optimize BufferedStream
+#   * Better documentation
+#   * Document MultipartParseError
+
+_ALLOWED_CONTENT_HEADERS = frozenset([
+    b'content-type',
+    b'content-disposition',
+    b'content-transfer-encoding',
+])
+
+DEFAULT_SUPPORTED_CHARSETS = (
```

Comment:
How was this list chosen? Might be worth an inline comment.

Reply:
The bad part is I don't remember now. :grimacing: 

Maybe we should just ditch this setting, and always allow all [standard encodings allowed by Python](https://docs.python.org/3.8/library/codecs.html#standard-encodings)?

Reply:
Yeah, I think that would work. It should be more than sufficient, and as long as our docs state this fact, there shouldn't be any big surprises for the developer.